File: token-validation-spring/src/main/java/no/nav/security/token/support/spring/api/EnableJwtTokenValidation.java
package no.nav.security.token.support.spring.api;
/*
 * THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
 * OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
 * ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A
 * PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
 */
import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Inherited;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.springframework.context.annotation.Import;

import no.nav.security.token.support.spring.EnableJwtTokenValidationConfiguration;

@Documented
@Inherited
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@Import({
    EnableJwtTokenValidationConfiguration.class
})
public @interface EnableJwtTokenValidation {
	String[] ignore() default {"org.springframework"};
}

File: token-validation-spring/src/main/kotlin/no/nav/security/token/support/spring/api/EnableJwtTokenValidation.kt
package no.nav.security.token.support.spring.api

import java.lang.annotation.Inherited
import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.annotation.AnnotationTarget.ANNOTATION_CLASS
import kotlin.annotation.AnnotationTarget.CLASS
import org.springframework.context.annotation.Import
import no.nav.security.token.support.spring.EnableJwtTokenValidationConfiguration

@MustBeDocumented
@Inherited
@Retention(RUNTIME)
@Target(ANNOTATION_CLASS, CLASS)
@Import(EnableJwtTokenValidationConfiguration::class)
annotation class EnableJwtTokenValidation(val ignore: Array<String> = ["org.springframework"])